network/network_linux_test.go

```diff
@@ -0,0 +1,98 @@
+package network
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/platform"
+)
+
+func TestAddDNSServersWithErr(t *testing.T) {
+	nm := &networkManager{
+		plClient: platform.NewMockExecClient(true),
+	}
+	_, err := nm.addDNSServers("invalid", []string{""})
+	if err == nil {
+		t.Errorf("Expected error")
+	}
+}
+
+func TestAddDNSServers(t *testing.T) {
+	testCases := []struct {
+		desc      string
+		osversion string
+		ifName    string
+		servers   []string
+		cmd       string
+		err       error
+	}{
+		{
```

Comment:
Do we know of any commonly used cmd(s) that should fail, so we can have some negative tests?